### DISCUSSION TITLE
Webreader zoom in and out

### DISCUSSION THREAD

**Discussion by wanestar** [gotson/komga#1596] [Ideas]
### Describe your suggested feature

Just a small thing, but I wish that you could set a zoom level for the comic while in webtoon mode. Either globally or per comic something like that... Preferably on the reader UI, for quick access while reading. 

I don't know how it's set up, but the Vertical modes "Screen" scale type would also work as it scales the page to what my zooming would do. Even having all the "Paginated Reader Options" in the "Webtoon Reader Options" would be a good addition I think.

I've been zooming in and out with chrome, but it gets annoying as it messes up the bottom bar, the home page, etc, and some other server pages I have running on the same IP. (chrome treats it as the same site and will zoom it out as well)

It's nothing major just an issue I think about a lot. 

### Other details

-I read on my PC on a 27-inch monitor. 
-It usually looks fine on a tablet, just the PC monitor gets unreadable at high resolutions 
-I only read from Komga's native reader. I like not going through other apps.

Or, possibly you could add mouse scroll support to the vertical mode as that automatically scales the pages into a readable position. I like to use the mouse instead of the arrow keys usually. 

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to the latest version.
- [X] I will fill out all of the requested information in this form.

**Comment by gotson** [comment]
Please explain with screenshots, I don't understand. If you have a 27" monitor the webtoon mode would align the width to your monitor width, it would be very wide. You want to zoom even further than that? Or is that to zoom out?

**Comment by wanestar** [comment]
Yes, sorry if it wasn't clear I want to be able to zoom out (and zoom in if the file is low resolution) natively. I read normal manga and comics in the webtoon mode, not webtoons, I just like the continuous scrolling of the mode with the mouse. 

1st screenshot is zoomed out to 25% with chrome 
2nd screenshot is the default zoom level at scale type: original 


![zoomed](https://user-images.githubusercontent.com/59542068/222934675-567f3968-7e1e-48ee-837d-155cd6336393.PNG)
![default](https://user-images.githubusercontent.com/59542068/222934677-f09b36ca-a7cc-4455-832a-456e5052b8c0.PNG)

**Comment by wanestar** [comment]
So to further add I've been experimenting with settings and figured out that the fit width mode will scale down the image when size padding is used...

That's on me for not figuring that out...

Still, I think that a zoom feature in the original mode could be considered as you cant zoom in and out on the fly in the " Fit Width " mode (Chrome zooming doesn't even work, it just resets the image back) and I like how the original mode keeps double spread pages original size as they are bigger and easier to read. But in the Fit Width mode, they are shrunk to a single page size.


Thanks and sorry for the confusion.

**Comment by gotson** [comment]
Webreader zoom was requested in #112 but never done

**Comment by BenMcLean** [comment]
I would normally scroll through comics in "fit width" mode but I also really need to be able to occasionally do a mouse wheel zoom in on specific words or things I can't make out sometimes because my eyesight is bad. (also pinch on touchscreens to do the same thing) CDisplayEx has this to provide a good example of it.

**Comment by titan-19** [comment]
Rather than requesting this feature you could use a browser add-on, I personally use [the zoom add-on](https://chromewebstore.google.com/detail/zoom-for-google-chrome/lajondecmobodlejlcjllhojikagldgd/reviews) in chrome you can use it to customize the manga width to whatever percentage you need it works perfectly for me.

**Comment by wanestar** [comment]
In the end, I ended up making a TamperMonkey script. Feel free to use it! It is fairly basic but it fits in with the UI.

https://github.com/wanestar/KomgaZoom

**Reply by kiekan88** [reply]
Would be really great to see this functionality added into Komga directly. Thanks for building this userscript, wanestar.

**Comment by kartikynwa** [comment]
In Firefox I have been able to emulate this by using the browser's native zooming (ctrl-plus, ctrl-minus). I think this should be a feature in the webreader. Have a slider for the "original" layout which allows scaling the images to a fixed percentage.
